lib/templatesimplifier.cpp

```diff
@@ -2344,6 +2344,27 @@ bool TemplateSimplifier::simplifyCalculations(Token* frontToken, Token *backToke
             tok->str(MathLib::toString(MathLib::toLongNumber(tok->str())));
         }
 
+        if (validTokenEnd(bounded, tok, backToken, 5) &&
+            Token::Match(tok, "decltype ( char|short|int|long { } )")) {
```

Comment:
Cant we simplify all `decltype ( %name% { } )`?

Reply:
Good catch.  Yes we can.